webapp/components/Password/Strength.vue

```diff
@@ -0,0 +1,131 @@
+<template>
+  <div class="field">
+    <div class="password-strength-meter">
+      <div
+        class="password-strength-meter-inner"
+        :class="'strength-' + strength"
+      />
+    </div>
+    <p class="help">
+      <span
+        v-if="pass"
+        :class="{ insecure: !isSecure }"
+      >
+        {{ $t('settings.security.change-password.passwordSecurity') }}:
+        <strong>{{ $t(`settings.security.change-password.passwordStrength${strength}`) }}</strong>
+      </span>
+      <span v-else>&nbsp;</span>
+    </p>
+  </div>
+</template>
+
+<script>
+import zxcvbn from 'zxcvbn'
+import { isEmpty } from 'lodash'
+
+export default {
+  name: 'PasswordMeter',
+  props: {
+    password: {
+      type: String,
+      required: true
+    }
+  },
+  data() {
+    return {
+      strength: null,
+      isSecure: false,
+      pass: this.password || null
+    }
+  },
+  watch: {
+    password(pass) {
+      // update password when prop is changing
+      this.pass = pass || null
+
+      // strength is the score calculated by zxcvbn
+      const strength = !isEmpty(this.pass) ? zxcvbn(this.pass).score : null
+      if (strength !== this.strength) {
+        this.strength = strength
+        this.isSecure = Boolean(strength >= 3)
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-09T00:27:03Z" title="Thursday, May 9th 2019, 2:27:03 am +02:00">May 9, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,139 @@
+<template>
+  <div class="field">
+    <div class="password-strength-meter">
+      <div
+        class="password-strength-meter-inner"
+        :class="strengthClass"
+      />
+    </div>
+    <p class="help">
+      <span
+        v-if="pass"
+        :class="{ insecure: (passwordStrength < 3) }"
+      >
+        {{ $t('settings.security.change-password.passwordSecurity') }}:
+        <strong>{{ $t(`settings.security.change-password.passwordStrength${passwordStrength}`) }}</strong>
+      </span>
+      <span v-else>&nbsp;</span>
+    </p>
+  </div>
+</template>
+
+<script>
+import zxcvbn from 'zxcvbn'
+import { isEmpty } from 'lodash'
+
+export default {
+  name: 'PasswordMeter',
+  props: {
+    password: {
+      type: String,
+      required: true
+    }
+  },
+  data() {
+    return {
+      lastScore: 0,
+      pass: this.password || null
+    }
+  },
+  computed: {
+    strengthClass() {
+      return `strength-${this.passwordStrength}`
+    }
+  },
+  watch: {
+    /**
+     * passwordStrength is the score calculated by zxcvbn
+     * @return {Number} Password Strength Score
+     */
+    passwordStrength() {
```


I moved this @ulfgebhardt because it was failing the linting
please see my commit message with a link to the SO post where I found the suggestion to move it to `watch`

Reply:
> [<img alt="ulfgebhardt" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-09T06:16:33Z" title="Thursday, May 9th 2019, 8:16:33 am +02:00">May 9, 2019</time>_

---


I Agree 100% - thanks for fixing this. As you can see i am still new to vue <3